src/controllers/API.js

```diff
@@ -73,6 +75,9 @@ class APIController {
             request.body.data, request.aboutRelationship
           );
 
+          // Camelize incoming request
+          parsedPrimary = formatters.camelizeResourceOrCollection(parsedPrimary, registry);
```

Comment:
For symmetry with my comment below about moving the dasherizing to `Document`, can we put this camelizing straight into the `parse-primary` step? Basically, I think of `parse-primary` as the the request formatting handler, and Document as the response formatter. At some point, they'll be broken out into more full-fledged components with better names...but for now I think we should keep all the logic within them.